src/swim/swim.c

```diff
@@ -705,28 +740,48 @@ swim_parse_message(struct swim_context *ctx, swim_id_t from,
 	enum swim_context_state ctx_state;
 	struct swim_member_state self_state;
 	swim_id_t self_id = swim_self_get(ctx);
-	swim_id_t id_target, id_sendto, to = SWIM_ID_INVALID;
+	swim_id_t id_target, id_sendto, to;
 	bool send_updates = false;
 	size_t i;
 	int rc = 0;
 
-	if (self_id == SWIM_ID_INVALID) /* not initialized yet */
+	if (self_id == SWIM_ID_INVALID || nupds == 0) /* not initialized yet */
 		return 0; /* Ignore this update */
 
+	if (d_log_check(DLOG_INFO | D_LOGFAC)) {
```

Comment:
this block seems similar to the one above - can it be combined into single function?

Reply:
Hmm, will extract...

Reply:
same comment as above

s/d_log_check(DLOG_INFO | D_LOGFAC)/D_LOG_ENABLED(DLOG_INFO)

Reply:
done.